packages/amplify_api_rest/lib/method_channel_api_rest.dart

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:collection';
+import 'dart:typed_data';
+import 'package:amplify_api_rest_plugin_interface/amplify_api_rest_plugin_interface.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'dart:convert';
+
+import 'amplify_api_rest.dart';
+
+const MethodChannel _channel =
+    MethodChannel('com.amazonaws.amplify/amplify_api_rest');
+
+/// An implementation of [AmplifyPlatform] that uses method channels.
+class AmplifyAPIRestMethodChannel extends AmplifyAPIRest {
+  RestOperation _restFunctionHelper(
+      {@required String methodName, @required RestOptions restOptions}) {
+    try {
+      // Send Request Code to Native
+      String code = UUID.getUUID();
+
+      Future<RestResponse> futureResponse =
+          _callNativeRestMethod(methodName, code, restOptions);
+
+      RestOperation restOperation = new RestOperation(
+          response: futureResponse, cancel: () => {cancelRequest(code)});
+
+      return restOperation;
+    } on PlatformException catch (e) {
+      throw (e);
+    }
```

Comment:
is this not a no-op?

Reply:
What do you mean by not a no-op? @Amplifiyer 

Are you pointing out that it shouldn't be necessary to surround the code here with try/catch as the error should be coming from _callNativeRestMethod.  Is that what you meant?

Reply:
yeah, since we are catching and throwing the same error and not doing anything in between, it seems this code is not really adding any value.

Reply:
ah good point, thank you!  In our merged branch these platformExceptions will be used to create an APIError via a formatError method